src/cmd.rs

```diff
@@ -77,6 +154,165 @@ pub fn osd_pool_set(cluster_handle: rados_t, pool: &str, key: &str, value: &str,
     Ok(())
 }
 
+pub fn osd_set(cluster_handle: rados_t, key: &str, force: bool, simulate: bool) -> Result<(), RadosError> {
+    let cmd = match force {
+        true => {
+            json!({
+                "prefix": "osd set",
+                "key": key,
+                "sure": "--yes-i-really-mean-it",
+            })
+        },
+        false => {
+            json!({
+                "prefix": "osd set",
+                "key": key,
+            })
+        },
+    };
+    debug!("osd set: {:?}", cmd.to_string());
+    if !simulate {
+        ceph_mon_command_without_data(cluster_handle, &cmd.to_string())?;
+    }
+    Ok(())
+}
+
+/// Possible values for the key are:
+/// full|pause|noup|nodown|noout|noin|nobackfill|norebalance|norecover|noscrub|
+/// nodeep-scrub|notieragent
+/// Check src/mon/MonCommands.h in the ceph github repo for more possible
+/// options
+pub fn osd_unset(cluster_handle: rados_t, key: &str, simulate: bool) -> Result<(), RadosError> {
+    let cmd = json!({
+        "prefix": "osd unset",
+        "key": key,
+    });
+    debug!("osd unset: {:?}", cmd.to_string());
+    if !simulate {
+        ceph_mon_command_without_data(cluster_handle, &cmd.to_string())?;
+    }
+    Ok(())
+}
+
+pub fn osd_tree(cluster_handle: rados_t) -> Result<CrushTree, RadosError> {
+    let cmd = json!({
+        "prefix": "osd tree",
+        "format": "json"
+    });
+    debug!("osd tree {:?}", cmd.to_string());
+    let result = ceph_mon_command_without_data(cluster_handle, &cmd.to_string())?;
+    if result.0.is_some() {
+        let return_data = result.0.unwrap();
+        let mut l = return_data.lines();
+        match l.next() {
+            Some(res) => return Ok(serde_json::from_str(res)?),
+            None => {
+                return Err(RadosError::Error(format!(
+                "Unable to parse osd tree output: {:?}",
+                return_data,
+            )))
+            },
+        }
+    }
+    Err(RadosError::Error("No response from ceph for osd tree".into()))
+}
+
+// Get cluster status
+pub fn status(cluster_handle: rados_t) -> Result<String, RadosError> {
+    let cmd = json!({
+        "prefix": "status",
+        "format": "json"
+    });
+    debug!("status {:?}", cmd.to_string());
+    let result = ceph_mon_command_without_data(cluster_handle, &cmd.to_string())?;
+    if result.0.is_some() {
+        let return_data = result.0.unwrap();
```

Comment:
aaaand crash

Reply:
I check to see if the result is_some() before I unwrap.  This could change to an if let instead or even a match because I need to do something either way.